infra/modules/service/scheduled-jobs.tf

```diff
@@ -0,0 +1,111 @@
+# docs: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_log_group
+resource "aws_cloudwatch_log_group" "scheduled_job" {
+  for_each = {
+    for index, job in var.scheduled_jobs :
+    job.name => job
+  }
+
+  name_prefix = "/aws/vendedlogs/states/${var.service_name}-${each.value.name}"
+
+  # Conservatively retain logs for 5 years.
+  # Looser requirements may allow shorter retention periods
+  retention_in_days = 1827
+
+  # TODO(https://github.com/navapbc/template-infra/issues/164) Encrypt with customer managed KMS key
+  # checkov:skip=CKV_AWS_158:Encrypt service logs with customer key in future work
+}
+
+# docs: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sfn_state_machine
+resource "aws_sfn_state_machine" "scheduled_job" {
+  for_each = {
+    for index, job in var.scheduled_jobs :
+    job.name => job
+  }
+
+  name     = "${var.service_name}-${each.value.name}"
+  role_arn = aws_iam_role.app_service.arn
+
+
+  definition = jsonencode({
+    "StartAt" : "ExecuteECSTask",
+    "States" : {
+      "ExecuteECSTask" : {
+        "Type" : "Task",
+        # docs: https://docs.aws.amazon.com/step-functions/latest/dg/connect-ecs.html
+        "Resource" : "arn:aws:states:::ecs:runTask.sync",
+        "Parameters" : {
+          "Cluster" : aws_ecs_cluster.cluster.arn,
+          "TaskDefinition" : aws_ecs_task_definition.app.arn,
+          "LaunchType" : "FARGATE",
+          "NetworkConfiguration" : {
+            "AwsvpcConfiguration" : {
+              "Subnets" : var.private_subnet_ids,
+              "SecurityGroups" : [aws_security_group.app.id],
+            }
+          },
+          "Overrides" : {
+            "ContainerOverrides" : [
+              {
+                "Name" : var.service_name,
+                "Environment" : each.value.environment
```

Comment:
@lorenyu I'm pretty sure it's an append action, rather than an override. Although I should test that.

That aside, you don't feel like there's a high chance that people will need to use environment variables to provide some extra configuration?

Reply:
I'm pretty sure it's an override since we have different ENV vars per job on MA PFML.
If a project ends up thinking that they'd like different environment variables then I'd like to better understand the use case and whether they're overengineering or not. I think most common cases won't need it and it adds unnecessary complexity around testing/quality assurance to test your code with various combinations of environment variables existing or not. For example, I've seen issues where all unit tests pass and most jobs pass but some jobs fail because an environment variable was missing in that one.

In the rare case that a project truly needs it they can add the functionality themselves, but I don't really want to encourage it.